### DISCUSSION TITLE
Automatic profile switching

### DISCUSSION THREAD

**Discussion by OPStellar** [pqrs-org/Karabiner-Elements#3626] [Ideas]
I use Karabiner for remapping the Right Shift key to a key that isn't on my keyboard. However, I only need this function for specific applications. Is there a way to automate profile switching for when specific applications are in the foreground?

Like, if I were to switch to a game application which uses WASD for directional input instead of the arrow keys, I want to set an automation which prompts Karabiner to arbitrarily swap to my "WASD arrows" profile which exchanges the arrow keys for WASD whenever I launch that particular application.

**Comment by CP20201041** [comment]
I use `frontmost_application_if` to detect when the specified app is in focus and then when any key is pressed (`"from": { "any": "key_code", "modifiers": { "optional": ["any"] } },`) i run a terminal command to switch profiles (`"to": [{ "shell_command": "/Library/Application\\ Support/org.pqrs/Karabiner-Elements/bin/karabiner_cli --select-profile 'PROFILE NAME'" }]`).
(capital letters for names to be replaced)
(bundle identifier for specific application can be found by using Karabiner):

```
{
    "description": "Auto Switch Profile",
    "manipulators": [
        {
            "conditions": [
                {
                    "bundle_identifiers": [
                        "BUNDLE IDENTIFIER"
                    ],
                    "type": "frontmost_application_if"
                }
            ],
            "from": {
                "any": "key_code",
                "modifiers": { "optional": ["any"] }
            },
            "to": [{ "shell_command": "/Library/Application\\ Support/org.pqrs/Karabiner-Elements/bin/karabiner_cli --select-profile 'PROFILE NAME'" }],
            "type": "basic"
        }
    ]
}
```
